### PR TITLE
udev-builtin: replace udev_device by sd_device

### PR DESCRIPTION
**Pull request by yuwata**
Continuation of recent works.

### CODE REVIEW

File: src/udev/udev-builtin-net_id.c [hunk 1]
```diff
@@ -147,98 +147,115 @@ struct netnames {
...
-        /* SMBIOS type 41 — Onboard Devices Extended Information */
-        if (!attr)
-                attr = udev_device_get_sysattr_value(names->pcidev, "index");
-        if (!attr)
-                return -ENOENT;
+        if (sd_device_get_sysattr_value(names->pcidev, "acpi_index", &attr) < 0) {
+                /* SMBIOS type 41 — Onboard Devices Extended Information */
+                r = sd_device_get_sysattr_value(names->pcidev, "index", &attr);
+                if (r < 0)
+                        return -ENOENT;
```
Comment (keszybz) [inline src/udev/udev-builtin-net_id.c]: `return r` ?

File: src/udev/udev-builtin-net_id.c [hunk 2]
```diff
@@ -147,98 +147,115 @@ struct netnames {
...
+        }
 
-        r = safe_atolu(attr, &idx);
-        if (r < 0)
-                return r;
+        idx = strtoul(attr, NULL, 0);
+        if (idx <= 0)
+                return -EINVAL;
```
Comment (keszybz) [inline src/udev/udev-builtin-net_id.c]: Hmm, why replace `safe_atolu` with `stroul`?

File: src/udev/udev-builtin-net_id.c [hunk 3]
```diff
@@ -443,21 +479,28 @@ static int names_vio(struct udev_device *dev, struct netnames *names) {
...
+        r = sd_device_get_parent(dev, &parent);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_subsystem(parent, &subsystem);
+        if (r < 0)
+                return r;
 
-        if (!streq_ptr("platform", udev_device_get_subsystem(parent)))
+        if (!streq("platform", subsystem))
```
Comment (keszybz) [inline src/udev/udev-builtin-net_id.c:497]: That's actually a yoda comparison, not that it matter too much. There's quite a few places like this, we'll probably want to normalize this at some point.

File: src/udev/udev-builtin-net_setup_link.c [hunk 4]
```diff
@@ -34,14 +34,18 @@ static int builtin_net_setup_link(struct udev_device *dev, int argc, char **argv
...
 
-        r = link_config_apply(ctx, link, dev->device, &name);
-        if (r < 0)
-                log_warning_errno(r, "Could not apply link config to %s, ignoring: %m", udev_device_get_sysname(dev));
+        r = link_config_apply(ctx, link, dev, &name);
+        if (r < 0) {
+                const char *sysname = NULL;
+
+                (void) sd_device_get_sysname(dev, &sysname);
+                log_warning_errno(r, "Could not apply link config to %s, ignoring: %m", sysname);
```
Comment (keszybz) [inline src/udev/udev-builtin-net_setup_link.c]: `strnull(sysname)`

File: src/udev/udev-builtin-path_id.c [hunk 5]
```diff
@@ -598,43 +598,47 @@ static int builtin_path_id(struct udev_device *dev, int argc, char *argv[], bool
...
+                return -EINVAL;
 
         /*
          * Do not return devices with an unknown parent device type. They
          * might produce conflicting IDs if the parent does not provide a
          * unique and predictable name.
          */
         if (!supported_parent)
-                return EXIT_FAILURE;
+                return -EINVAL;
```
Comment (keszybz) [inline src/udev/udev-builtin-path_id.c]: The return convention is updated. I think this should be mentioned in the commit message.

It seems that the return value from `udev_builtin_run` is generally ignored almost everywhere. In `udevadm-test-builtin.c` the check is only for negative values.

File: src/udev/udev-builtin-usb_id.c [hunk 6]
```diff
@@ -357,33 +361,36 @@ static int builtin_usb_id(struct udev_device *dev, int argc, char *argv[], bool
...
-                const char *usb_model = NULL;
+                const char *usb_model;
 
-                usb_model = udev_device_get_sysattr_value(dev_usb, "product");
-                if (!usb_model)
+                if (sd_device_get_sysattr_value(dev_usb, "product", &usb_model) < 0)
                         usb_model = product_id;
                 if (!usb_model)
-                        return EXIT_FAILURE;
+                        return -ENOENT;
```
Comment (keszybz) [inline src/udev/udev-builtin-usb_id.c]: Here too, please add a sentence to the commit message about the return convention change.

File: src/udev/udev-builtin-blkid.c [hunk 7]
```diff
@@ -210,14 +210,13 @@ static int probe_superblocks(blkid_probe pr) {
...
         const char *devnode, *root_partition = NULL, *data, *name;
         _cleanup_(blkid_free_probep) blkid_probe pr = NULL;
         bool noraid = false, is_gpt = false;
         _cleanup_close_ int fd = -1;
         int64_t offset = 0;
         int nvals, i, r;
-        sd_device *dev = _dev->device;
 
         static const struct option options[] = {
                 { "offset", required_argument, NULL, 'o' },
```
Comment (keszybz) [inline src/udev/udev-builtin-blkid.c:224]: [This comment is really for line 252:]

Here `EXIT_FAILURE` is kept. I think the return convention needs to be changed everywhere.

File: src/udev/udev-builtin.c [hunk 8]
```diff
@@ -90,14 +90,16 @@ bool udev_builtin_run_once(enum udev_builtin_cmd cmd) {
...
         enum udev_builtin_cmd i;
-        char *pos;
+        _cleanup_free_ char *name = NULL;
+
+        assert(command);
+
+        command += strspn(command, WHITESPACE);
+        name = strndup(command, strchrnul(command, ' ') - command);
+        if (!name)
+                return UDEV_BUILTIN_MAX;
```
Comment (keszybz) [inline src/udev/udev-builtin.c]: Is this `strndup` actually needed? Maybe calculate the length and use `strneq` below?

**Comment by yuwata** [comment]
@keszybz Thank you for your review. I hope all your comments are addressed. Note,
- All return values are changed to negative `errno`. See the last commit.
- `streq_ptr()` is replaced by `streq()` if it is applicable.

PTAL.

**Comment by keszybz** [comment]
```
Oct 23 10:55:55 autopkgtest systemd[1]: Failed to start Light Display Manager.
FAIL
```

**Pull request merged**